examples/chip-tool/py_matter_chip_tool_adapter/matter_chip_tool_adapter/decoder.py

```diff
@@ -252,6 +253,42 @@ def maybe_convert(self, typename: str, value):
         return value
 
 
+class DarwinAnyFormatConverter(BaseConverter):
+    """
+    Darwin payloads format for *ById commands is different from the base
+    format used for other commands.
+    """
+
+    def run(self, specs, value, cluster_name: str, typename: str, array: bool):
+        if isinstance(value, list) and len(value) >= 1 and isinstance(value[0], dict) and value[0].get('data') is not None:
+            value = [self.__convert_data_to_value(
+                specs, item_value, cluster_name, typename) for item_value in value]
+        return value
+
+    def __convert_data_to_value(self, specs, value, cluster_name: str, typename: str):
+        if not isinstance(value, dict):
+            return value
+
+        data = value.get('data')
+        if data:
+            value = data.get('value')
+            value_type = data.get('type')
+
+            if isinstance(value, list):
+                if value_type == 'Structure':
+                    struct = {}
+                    for field in value:
+                        context_tag = field.get('contextTag')
+                        struct[str(context_tag)] = self.__convert_data_to_value(
+                            specs, field, cluster_name, typename)
```

Comment:
Is using `typename` here really the right thing?  Or should it be the name of the type of the field?

The fact that this is very unclear suggests this method needs more docs on the arguments...

Reply:
Apologies for that!
I was a bit overzealous with the arguments. We could simplify it by passing a single argument instead of four, which should make it clearer.